src/test/kotlin/io/libp2p/pubsub/PubsubApiTest.kt

```diff
@@ -0,0 +1,72 @@
+package io.libp2p.pubsub
+
+import io.libp2p.core.crypto.PrivKey
+import io.libp2p.core.pubsub.MessageApi
+import io.libp2p.core.pubsub.PubsubPublisherApi
+import io.libp2p.core.pubsub.Subscriber
+import io.libp2p.core.pubsub.Topic
+import io.libp2p.etc.types.toByteArray
+import io.libp2p.etc.types.toByteBuf
+import io.libp2p.etc.types.toBytesBigEndian
+import io.libp2p.etc.types.toProtobuf
+import io.libp2p.pubsub.flood.FloodRouter
+import io.netty.buffer.ByteBuf
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import pubsub.pb.Rpc
+import java.nio.charset.StandardCharsets
+import java.time.Duration
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.TimeUnit
+
+class PubsubApiTest {
+
+    @Test
+    fun testNoFromMessageField() {
+        class CustomApi(router: PubsubRouter) : PubsubApiImpl(router) {
+            inner class CustomPublisher(privKey: PrivKey, seqId: Long) : PublisherImpl(privKey, seqId) {
+                override fun createMessageToSign(data: ByteBuf, vararg topics: Topic): Rpc.Message {
+                    // not including 'from' field
```

Comment:
why?

Reply:
`From` field is subject to be removed at some point in the spec